packages/dds/tree/src/feature-libraries/sequence-field/compose.ts

```diff
@@ -422,53 +373,65 @@ function applyMoveEffects<TNodeChange>(
     moveEffects: MoveEffectTable<TNodeChange>,
 ): MarkList<TNodeChange> {
     const factory = new MarkListFactory<TNodeChange>(moveEffects);
-    for (const mark of marks) {
-        const splitMarks = applyMoveEffectsToMark(
-            mark,
-            undefined,
-            moveEffects,
-            () => fail("Should not generate IDs"),
-            false,
-            // TODO: Should pass in revision for new changes
-            (a, b) => composeChildChanges(a, b, undefined, composeChild),
-        );
+    const queue = new MarkQueue(
+        marks,
+        undefined,
+        moveEffects,
+        () => fail("Should not generate IDs"),
+        false,
+        // TODO: Should pass in revision for new changes
+        (a, b) => composeChildChanges(a, b, undefined, composeChild),
+    );
 
-        for (const splitMark of splitMarks) {
-            factory.push(splitMark);
-        }
+    while (!queue.isEmpty()) {
+        factory.push(queue.dequeue() as Mark<TNodeChange>);
```

Comment:
Nitpicking here, but I find this less likely to hide bugs if the rest of the code shifts from under it:
```suggestion
        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
        factory.push(queue.dequeue()!);
```